pkg/console/subresource/deployment/deployment.go

```diff
@@ -84,6 +84,9 @@ func DefaultDeployment(operatorConfig *operatorv1.Console, cm *corev1.ConfigMap,
 	gracePeriod := int64(40)
 	tolerationSeconds := int64(120)
 	volumeConfig := defaultVolumeConfig()
+	if routesub.IsCustomRouteSecretSet(operatorConfig) {
+		volumeConfig = useCustomServingCertSecret(operatorConfig.Spec.Route.Secret.Name, volumeConfig)
```

Comment:
why are you mounting this? The contents are supposed to be used in the route's `spec.tls.termination.{cert,key,caCertificate}` fields

Reply:
Also, you should have a hard requirement on the secret you're syncing to contain all of `tls.crt`, `tls.key` and `ca.crt` fields

Reply:
> why are you mounting this? The contents are supposed to be used in the route's spec.tls.termination.{cert,key,caCertificate} fields

I though that the secret needs to be mounted into the `console` pod.

> why are you mounting this? The contents are supposed to be used in the route's spec.tls.termination.{cert,key,caCertificate} fields

also `ca.crt`?
all of them needs to be used in the route's `spec.tls.termination.{cert,key,caCertificate}` fields?

Reply:
I think you would want to have the CA certificate there as well to prevent issues that the routing team has - https://github.com/openshift/cluster-ingress-operator/pull/329. Generally, it would make your life easier.

Also, you should validate the secret and don't progress further if it's invalid.